lib/LocalVerifiableCredentialStore.js

```diff
@@ -142,4 +170,58 @@ export class LocalVerifiableCredentialStore {
     }
     return typeof issuer === 'string' ? issuer : issuer.id;
   }
+
+  async _queryByExample({credentialQuery}) {
+    if(!credentialQuery) {
+      throw new Error(
+        '"credentialQuery" is needed to execute a QueryByExample.');
+    }
+    if(typeof credentialQuery !== 'object') {
+      throw new Error('"credentialQuery" must be an object or an array.');
+    }
+
+    // normalize query to be an array
+    let query;
+    if(Array.isArray(credentialQuery)) {
+      query = credentialQuery;
+    } else {
+      query = [credentialQuery];
+    }
+
+    const _query = async ({example}) => {
+      const {type, trustedIssuer = []} = example;
+      // normalize trusted issuers to be an array
+      let trustedIssuers;
+      if(Array.isArray(trustedIssuer)) {
+        trustedIssuers = trustedIssuer;
+      } else {
+        trustedIssuers = [trustedIssuer];
+      }
+
+      // build query to find all VCs that match any combination of type+issuer
+      const query = [];
+      const issuers = trustedIssuers.filter(({required}) => required)
+        .map(({issuer}) => issuer);
+      const types = Array.isArray(type) ? type : [type];
+      for(const type of types) {
+        if(issuers.length === 0) {
+          query.push({type});
+          continue;
+        }
+        for(const issuer of issuers) {
+          query.push({type, issuer});
+        }
+      }
+      const queryType = query.map(q => q.type);
+      return this.find({type: queryType});
+    };
+
+    // only look for credentials that are required
+    const requiredCredentials = await Promise.all(query.map(_query));
```

Comment:
This is already setup to use the p-map API, might as well go ahead and use it to limit concurrency.  https://github.com/sindresorhus/p-map#pmapinput-mapper-options

Promise.all can create problems with arrays of unbounded size.  There's a possibility that there could be large lists of credentials.